tools/utils/array_readwriter/csv.py

```diff
@@ -0,0 +1,28 @@
+import logging
+import pandas as pd
+import pyarrow
+import pyarrow.csv
+from .registry import register
+
+@register("csv")
+class CSV(object):
+    def __init__(self, delimiter=','):
+        self.delimiter = delimiter
+
+    def read(self, path):
+        logging.info('Reading from %s using CSV format with configuration %s' % (
+            path, self.__dict__))
+        # do not read the first line as header
+        read_options = pyarrow.csv.ReadOptions(autogenerate_column_names=True)
+        parse_options = pyarrow.csv.ParseOptions(delimiter=self.delimiter)
+        arr = pyarrow.csv.read_csv(path, read_options=read_options, parse_options=parse_options)
+        logging.info('Done reading from %s' % path)
+        return arr.to_pandas().to_numpy()
+
+    def write(self, path, arr):
+        logging.info('Writing to %s using CSV format with configuration %s' % (
+            path, self.__dict__))
+        write_options = pyarrow.csv.WriteOptions(include_header=False, delimiter=self.delimiter)
+        arr = pyarrow.Table.from_pandas(pd.DataFrame(arr))
```

Comment:
pyarrow doesn't work with numpy array?

Reply:
I don't think pyarrow Tables can directly convert to numpy arrays.  https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table